cypress/integration/brewery/ScenarioSharing.js

```diff
@@ -1,40 +1,211 @@
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.ADMIN).click();
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VALIDATOR).should('not.be.checked');
+    for (let i = 0; i < 2; ++i) grantedPermissions.push(notGrantedPermissions.shift());
+    RolesEdition.checkShareScenarioDialogGrantedPermissionChip(grantedPermissions, notGrantedPermissions);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VIEWER).click();
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VIEWER).should('be.checked');
+    RolesEdition.getShareScenarioDialogConfirmAddAccessButton().click();
+
+    RolesEdition.addAgent(USERS_LIST[2].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.EDITOR).click();
+    RolesEdition.getShareScenarioDialogConfirmAddAccessButton().click();
+
+    RolesEdition.addAgent(USERS_LIST[3].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VALIDATOR).click();
+    RolesEdition.getShareScenarioDialogConfirmAddAccessButton().click();
+
+    const expectedNewPermissions = {
+      ACLSecurity: [
+        { id: USERS_LIST[1].email, role: ROLES.SCENARIO.VIEWER },
+        { id: USERS_LIST[2].email, role: ROLES.SCENARIO.EDITOR },
+        { id: USERS_LIST[3].email, role: ROLES.SCENARIO.VALIDATOR },
+      ],
+    };
+    RolesEdition.confirmNewPermissions(expectedNewPermissions);
+
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[1].email).should('have.text', USERS_LIST[1].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('input').should('value', ROLES.SCENARIO.VIEWER);
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[2].email).should('have.text', USERS_LIST[2].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[2].email).find('input').should('value', ROLES.SCENARIO.EDITOR);
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[3].email).should('have.text', USERS_LIST[3].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[3].email).find('input').should('value', ROLES.SCENARIO.VALIDATOR);
   });
 
-  it('is shown a message error when unauthorized changes are tried in scenario sharing pop-up', () => {
-    // TODO: open share pop-up and check error messages are displayed when unauthorized changes are made:
-    //  - remove the last admin of a scenario
-    //  - ...
+  it('Check Viewer permissions', () => {
+    Scenarios.getScenarioValidateButton().should('not.exist');
+
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getShareScenarioDialogAgentsSelect().should('not.exist');
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[1].email).should('have.text', USERS_LIST[1].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('input').should('value', ROLES.SCENARIO.VIEWER);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('div').should('have.attr', 'aria-disabled');
```

Comment:
Can't we check with `should('be.disabled')` instead of checking `aria-disabled` ? If it can't be done easily, then we should probably create a generic action `isACLSelectorEnabled` to factorize it

Reply:
No, the component don't use _enabled_ attribute and then _should('be.disabled')_ don't work.